packages/wallet-sdk/src/core/type/index.ts

```diff
@@ -1,5 +1,33 @@
 // Copyright (c) 2018-2024 Coinbase, Inc. <https://www.coinbase.com/>
 
+import { ProviderInterface } from './ProviderInterface';
+
+export interface Window {
+  top: Window;
+  ethereum?: ProviderInterface;
+  coinbaseWalletExtension?: ProviderInterface;
+}
+
+export function fetchCoinbaseInjectedProvider(
```

Comment:
let's move this to https://github.com/coinbase/coinbase-wallet-sdk/blob/master/packages/wallet-sdk/src/core/eip1193Utils.ts (we might rename this file though)